### ISSUE TITLE
allow dot-notated keys for model attributes of type json

### ISSUE THREAD

**Issue by japel** [balderdashy/sails-mongo#280]
The JSON type seems perfect to store configuration objects, but I'm stuck here, because lots of configurations I would like to store (like elasticsearch query configs, etc) use dot notation in keys.

If I pass:
```json
{
    "my.awesome.dot.notated.json.key": true
}
```
to jsonlint.com it tells me it is perfectly valid JSON

I found that the check is done @ js-bson
https://github.com/mongodb/js-bson/blob/master/lib/bson/bson.js

Apparently it is possible to pass {checkKeys: true/false} to that, so it is possible to use dots and dollars in json keys.

But sails-mongo does not pass any options.

So now the question is:
Is there a reason why we cannot use dots in json keys?
If yes: which is the reason?
If no: would it be safe change sails-mongo to pass {checkKeys: true/false} when doing a model.create()?

Thanks in advance

japel

**Comment by sailsbot** [comment]
Thanks for posting, @japel.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
